contracts/APM.sol

```diff
@@ -66,46 +66,43 @@ contract APM is IAPM, GovernanceOwnable {
         returns (uint256 reserveA, uint256 reserveB)
     {
         (reserveA, reserveB) = (
-            getReservesOneToken(tokenA, tokenB),
-            getReservesOneToken(tokenB, tokenA)
+            _getReservesOneToken(tokenA, tokenB),
+            _getReservesOneToken(tokenB, tokenA)
         );
     }
 
-    function updateWhenAddLiquidityOneToken(
+    function _updateWhenAddLiquidityOneToken(
         uint256 amountA,
         address tokenA,
         address tokenB
     ) private {
-        UpdateData memory updateData;
-        updateData.amountA = amountA;
-        updateData.tokenA = tokenA;
-        updateData.tokenB = tokenB;
 
-        uint256 totalReserveA = totalReserve[updateData.tokenA]; //gas saving
+        uint256 totalReserveA = totalReserve[tokenA]; //todo : should be put totalreserve[tokenA] or IERC20(tokenA).balanceOf(address(this)) ?
```

Comment:
yes, totalReserve[A] is good, because if someone transfers tokens to APM, u won't be able to track it... meaning we might have more tokens in the balance of APM than in the total reserve variable but never the contrary